0029-stream/0029-stream.md

```diff
@@ -174,7 +174,11 @@ Implementations SHOULD wait for a valid response (encrypted with the same shared
 
 #### 4.3.1. Connection Asset Details
 
-Each endpoint MAY expose their asset details by sending a `ConnectionAssetDetails` frame.
+Each endpoint MAY expose their asset details by sending a `ConnectionAssetDetails` frame. This frame is optional because some use-cases, such as Web Monetization, do not require it.
+
+Asset details exposed by this frame MUST not change during the lifetime of a Connection. Because of this, endpoints that honor `ConnectionAssetDetails` frames SHOULD ignore any repeat `ConnectionAssetDetails` frames for a given Connection.
+
+Endpoints that wish to discover the other endpoint's asset details MAY send a `ConnectionNewAddress` frame to prompt the endpoint to respond with a `ConnectionAssetDetails` frame.
```

Comment:
Alternatively, could we change this so sending a `ConnectionAssetDetails` frame triggers the other endpoint to respond with a `ConnectionAssetDetails` frame? That way it'd be more symmetric ("I'll tell you my asset details if you tell me yours").

I know I proposed this 😆...but bundling this behavior to the `ConnectionNewAddress` frame does seem somewhat odd (e.g. how are asset details related to changing your address mid-connection?)

Reply:
Or we could move this out of STREAM entirely and make it part of SPSP?

I feel like we may have explored that in the past but I can't recall

Reply:
> Or we could move this out of STREAM entirely and make it part of SPSP?

I've had various discussions recently about about potentially deprecating the `ConnectionAssetDetails` frame in STREAM.

Happy to be argued out of this position, but currently I don't prefer this because it forces whatever gets used instead of SPSP to have to provide `asset_code` and `asset_scale`, which _might_ be undesirable. 

E.g., imagine an alternative to SPSP that enables a sender and receiver who have advertised their public keys to arrive at a derived shared secret and receiver address. Forcing this future protocol to _also_ define some endpoint to privately get the asset details seems unnecessary since  these can be obtained privately inside of a Stream Connection.

Reply:
> could we change this so sending a ConnectionAssetDetails frame triggers the other endpoint to respond with a ConnectionAssetDetails frame

I actually like this a lot. Originally I wanted to avoid a circular loop, but as long as the response-frame is sent back in the fulfill/reject packet, I think that can work, and it's a nicer, more generic querying mechanism for getting the other endpoint's asset details.

Reply:
> Or we could move this out of STREAM entirely and make it part of SPSP?

How would the sender inform the recipient of their asset details? Query params?

Reply:
> Or we could move this out of STREAM entirely and make it part of SPSP?

:thinking: https://github.com/interledgerjs/ilp-protocol-stream/pull/69 @adrianhopebailie